pkg/errors/armerrors_test.go

```diff
@@ -0,0 +1,146 @@
+package errors
+
+
+import (
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"testing"
+
+    "bytes"
+    "io"
+    "net/http"
+
+)
+
+func TestIsNotFoundErr(t *testing.T) {
+	err1 := &azcore.ResponseError{ErrorCode: ResourceNotFound}
+	if !IsNotFoundErr(err1) {
+		t.Errorf("Expected IsNotFoundErr to return true for err1, but it returned false")
+	}
+
+	err2 := &azcore.ResponseError{ErrorCode: "SomeOtherErrorCode"}
+	if IsNotFoundErr(err2) {
+		t.Errorf("Expected IsNotFoundErr to return false for err2, but it returned true")
+	}
+
+	if IsNotFoundErr(nil) {
+		t.Errorf("Expected IsNotFoundErr to return false for nil input, but it returned true")
+	}
+}
+
+
+
+type testCase struct {
+    description        string
+    responseError      *azcore.ResponseError
+    expected bool
+}
+
+func TestSKUFamilyQuotaHasBeenReached(t *testing.T) {
+    testCases := []testCase{
+        {
+            description: "Quota Exceeded",
+            responseError: &azcore.ResponseError{
+                ErrorCode:   "OperationNotAllowed",
+                StatusCode: http.StatusForbidden,
+                RawResponse: &http.Response{
+                    Body: io.NopCloser(bytes.NewReader([]byte(`{"error": {"code": "OperationNotAllowed", "message": "Family Cores quota exceeded"}}`))),
+                },
+            },
+            expected: true,
+        },
+        {
+            description: "Different Error Code",
+            responseError: &azcore.ResponseError{
+                ErrorCode:   "ResourceNotFound",
+                StatusCode: http.StatusNotFound,
+                RawResponse: &http.Response{
+                    Body: io.NopCloser(bytes.NewReader([]byte(`{"error": {"code": "ResourceNotFound"}}`))),
+                },
+            },
+            expected: false,
+        },
+    }
+
+    for _, tc := range testCases {
+        t.Run(tc.description, func(t *testing.T) {
+            if got := SKUFamilyQuotaHasBeenReached(tc.responseError); got != tc.expected {
+                t.Errorf("SKUFamilyQuotaHasBeenReached() = %v, want %v", got, tc.expected)
```

Comment:
We should look at not using `%v` or `%+v` whenever possible. 

It can be a risk of leaking info to the logs. 

Here: `%t` should work for both bools I believe